GP/toy_experiments/toy_sin_noise.py

```diff
@@ -1,54 +1,50 @@
-# Author: Ryan-Rhys Griffiths
+# Author: Ryan-Rhys Griffiths (Modified by Alex Aldrick)
 """
-This module contains the code for benchmarking heteroscedastic Bayesian Optimisation on a number of toy functions.
+This module contains the code for benchmarking heteroscedastic Bayesian Optimisation on a 1D toy problem.
 """
 
 import matplotlib.pyplot as plt
 from matplotlib.ticker import MaxNLocator
 import numpy as np
+from numpy.core._multiarray_umath import ndarray
 
 from acquisition_functions import heteroscedastic_expected_improvement, heteroscedastic_propose_location, \
-    my_propose_location, my_expected_improvement, augmented_expected_improvement, heteroscedastic_augmented_expected_improvement
+    my_propose_location, my_expected_improvement, augmented_expected_improvement, \
+    heteroscedastic_augmented_expected_improvement
 from objective_functions import linear_sin_noise, max_sin_noise_objective
 
-
 if __name__ == '__main__':
 
     modification = True  # Switches between sin(x) - False and sin(x) + 0.05x - True
     coefficient = 0.2  # tunes the relative size of the maxima in the function (used when modification = True)
 
     # Number of iterations
+    random_trials = 10
     bayes_opt_iters = 5
+    random_sampling_iters = 5
 
     # We perform random trials of Bayesian Optimisation
 
+    rand_running_sum = np.zeros(random_sampling_iters)
+    rand_squares = np.zeros(random_sampling_iters)
     homo_running_sum = np.zeros(bayes_opt_iters)
-    homo_squares = np.zeros(bayes_opt_iters)  # Following the single-pass estimator given on pg. 192 of mathematics for machine learning
+    homo_squares = np.zeros(
+        bayes_opt_iters)  # Following the single-pass estimator given on pg. 192 of mathematics for machine learning
     hetero_running_sum = np.zeros(bayes_opt_iters)
     hetero_squares = np.zeros(bayes_opt_iters)
-    aug_running_sum = np.zeros(bayes_opt_iters)
-    aug_squares = np.zeros(bayes_opt_iters)
-    aug_het_running_sum = np.zeros(bayes_opt_iters)
-    aug_het_squares = np.zeros(bayes_opt_iters)
 
     # We compute the objective corresponding to aleatoric noise only
 
     homo_noise_running_sum = np.zeros(bayes_opt_iters)
-    homo_noise_squares = np.zeros(bayes_opt_iters)  # Following the single-pass estimator given on pg. 192 of mathematics for machine learning
+    homo_noise_squares = np.zeros(
+        bayes_opt_iters)  # Following the single-pass estimator given on pg. 192 of mathematics for machine learning
     hetero_noise_running_sum = np.zeros(bayes_opt_iters)
     hetero_noise_squares = np.zeros(bayes_opt_iters)
-    aug_noise_running_sum = np.zeros(bayes_opt_iters)
-    aug_noise_squares = np.zeros(bayes_opt_iters)
-    aug_het_noise_running_sum = np.zeros(bayes_opt_iters)
-    aug_het_noise_squares = np.zeros(bayes_opt_iters)
-
-    random_trials = 10
 
-    for i in range(random_trials):
+    for i in range(random_trials): #random trials to get average across x number of trials
```

Comment:
We can put the random sampling in the same loop as the BayesOpt strategies because they'll both run for the same number of iterations.